qcoro/qcorotask.h

```diff
@@ -742,23 +772,40 @@ inline T waitFor(QCoro::Task<T> &&task) {
  *        For void tasks, it needs to have no arguments.
  *        For all other types, it takes a value of the type as single argument.
  */
-template <typename T, typename Callback>
-requires std::is_invocable_v<Callback> || std::is_invocable_v<Callback, T>
-void connect(QCoro::Task<T> &task, QObject *context, Callback func) {
+template <typename T, typename QObjectSubclass, typename Callback>
+requires std::is_invocable_v<Callback> || std::is_invocable_v<Callback, T> || std::is_invocable_v<Callback, QObjectSubclass>
```

Comment:
Shouldn't it be `std::is_invocable_v<Callback, QObjectSubclass *>` ? Since the argument is `QObjectSublass *context`, then `QObjectSubclass` type is not a pointer type, but the callback will definitely be taking the QObject as a pointer?

Reply:
I think the reason this compiles is because we don't actually have a testcase that covers the case when Callback is a member function. I'd like to have one, please, to make sure we don't accidentally break this case in the future.

Reply:
```C++
        QCoro::connect(timer(), this, &QCoroTaskTest::callbackCalled);
```

was supposed to be a testcase for this, interesting that it compiles.